src/Chapter2.hs

```diff
@@ -857,7 +874,9 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind [] = []
+rewind (a : as) = rewind as ++ [a]
```

Comment:
Your solution is correct! However, it is slow. In lists, it is quite slow to add anything at the end of the list. That is why it is always better to rewrite it with the `:` cons. Remember the explanation with trains? 🚂 🚋 🚋 

That is why a more efficient solution is with the accumulator and the recursive function that will do the addition at the start of the list which is instant!

You can read a bit more about the `go` pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go

Reply:
Ah, good point. I already suspected that the solution wasn't ideal in terms of performance, but I couldn't come up with a better solution in the first place. With your hint, I found a variant using the `go` pattern :smiley:

Reply:
This is the absolutely right solution you have come up with for the `reverse` function and go pattern 👍🏼 Feel free to merge this PR 🚢